browser/ui/views/playlist/playlist_action_dialogs.h

```diff
@@ -0,0 +1,135 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_UI_VIEWS_PLAYLIST_PLAYLIST_ACTION_DIALOGS_H_
+#define BRAVE_BROWSER_UI_VIEWS_PLAYLIST_PLAYLIST_ACTION_DIALOGS_H_
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "brave/browser/playlist/playlist_tab_helper_observer.h"
+#include "brave/browser/ui/views/playlist/selectable_list_view.h"
+#include "brave/components/playlist/common/mojom/playlist.mojom.h"
+#include "chrome/browser/ui/views/frame/browser_view.h"
+#include "components/constrained_window/constrained_window_views.h"
+#include "ui/base/metadata/metadata_header_macros.h"
+#include "ui/views/controls/textfield/textfield_controller.h"
+#include "ui/views/window/dialog_delegate.h"
+
+namespace views {
+class BoxLayoutView;
+}  // namespace views
+
+namespace playlist {
+class PlaylistTabHelper;
+}  // namespace playlist
+
+// Base class for playlist action dialogs. Responsible for creating widget
+// and anchoring.
+class PlaylistActionDialog : public views::DialogDelegateView {
+ public:
+  METADATA_HEADER(PlaylistActionDialog);
+
+  template <class Dialog, typename... Args>
+  static void Show(BrowserView* browser_view, Args&&... args) {
+    DCHECK(browser_view);
+    auto* browser_widget = browser_view->GetWidget();
+    DCHECK(browser_widget);
+
+    auto dialog = std::make_unique<Dialog>(typename Dialog::PassKey(),
+                                           std::forward<Args>(args)...);
+    dialog->SetModalType(ui::ModalType::MODAL_TYPE_WINDOW);
```

Comment:
As this dialog is launched from current tab's action, I think tab modal is more appropriate?
I don't have strong opinion abou this. Just curious why window modal is choosen here.

Reply:
I'd like to put off the decision for now. As some of these dialogs could be triggered from sidebar's web ui, the action dialog might not mapped with the current active tab. But I'd prefer window modal if possible, so we'll see 😄